### PR TITLE
CLOUDP-100237: Propose homebrew update command on version update alert

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Display instructions on how to update via homebrew if installed via homebrew.

_Jira ticket:_ CLOUDP-100237

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/root/builder.go [hunk 1]
```diff
@@ -139,18 +144,49 @@ func (opts *BuilderOpts) hasNewVersionAvailable() (newVersionAvailable bool, new
...
 }
 
+func isAtLeast24HoursPast(t time.Time) bool {
+	return !t.IsZero() && time.Since(t) >= time.Hour*24
+}
+
+func isHomebrew() bool {
+	if runtime.GOOS != "darwin" {
+		return false
+	}
```
Comment (gssbzn) [inline internal/cli/root/builder.go:166]: homebrew support linux, any particular reason to skip it?

File: internal/cli/root/builder.go [hunk 2]
```diff
@@ -139,18 +144,49 @@ func (opts *BuilderOpts) hasNewVersionAvailable() (newVersionAvailable bool, new
...
+	brewPrefixPath := strings.TrimSpace(string(brewPrefixBytes))
+	brewBinPrefix := filepath.Join(brewPrefixPath, "bin") + string(filepath.Separator)
+
+	mongocliPath, err := os.Executable()
+
+	if err != nil {
+		return false
+	}
+
+	return strings.HasPrefix(mongocliPath, brewBinPrefix)
```
Comment (gssbzn) [inline internal/cli/root/builder.go:187]: [q] wouldn't you be able to use only `brewPrefixPath` here? asking if we can avoid the appending of `bin`

File: internal/cli/root/builder.go (hunk 1 above)
Comment (fmenezes) [reply internal/cli/root/builder.go:166]: my bad

**Review by colm-quinn — APPROVED** [review]
LGTM on changes since last review - not too sure on the internals of homebrew, so this is mostly a code check for anything obvious. You may want another review on the homebrew specific points.

**Pull request merged**